src/lib/SX127xDriver/SX127x.cpp

```diff
@@ -289,6 +327,12 @@ void ICACHE_RAM_ATTR SX127xDriver::TXnb()
 void ICACHE_RAM_ATTR SX127xDriver::RXnbISR()
 {
   hal.readRegisterFIFO(RXdataBuffer, PayloadLength);
+  if (timeoutSymbols)
+  {
+    // From page 42 of the datasheet rev 7
+    // In Rx Single mode, the device will return to Standby mode as soon as the interrupt occurs
+    currOpmode = SX127x_OPMODE_STANDBY;
```

Comment:
read radio state from reported status byte instead:
````
    //currOpmode = SX1280_MODE_FS;
    currOpmode = (SX1280_RadioOperatingModes_t)(
        (status & SX1280_STATUS_MODE_MASK) >> SX1280_STATUS_MODE_SHIFT);
```
Use `GetIrqStatus` to update status byte...

Reply:
Status byte content:

```
/* Status byte flags */
enum
{
    SX1280_STATUS_BUSY = 0x01,
    // 4:2 = Command status
    SX1280_STATUS_CMD_STATUS_SHIFT          = 2,
    SX1280_STATUS_CMD_STATUS_MASK           = 0x7 << SX1280_STATUS_CMD_STATUS_SHIFT,
    SX1280_STATUS_CMD_STATUS_CMD_OK         = 0x1 << SX1280_STATUS_CMD_STATUS_SHIFT,
    SX1280_STATUS_CMD_STATUS_DATA_AVAILABLE = 0x2 << SX1280_STATUS_CMD_STATUS_SHIFT,
    SX1280_STATUS_CMD_STATUS_CMD_TIMEOUT    = 0x3 << SX1280_STATUS_CMD_STATUS_SHIFT,
    SX1280_STATUS_CMD_STATUS_CMD_ERROR      = 0x4 << SX1280_STATUS_CMD_STATUS_SHIFT,
    SX1280_STATUS_CMD_STATUS_EXEC_FAIL      = 0x5 << SX1280_STATUS_CMD_STATUS_SHIFT,
    SX1280_STATUS_CMD_STATUS_TX_DONE        = 0x6 << SX1280_STATUS_CMD_STATUS_SHIFT,
    // 7:5 = Circuit mode
    SX1280_STATUS_MODE_SHIFT        = 5,
    SX1280_STATUS_MODE_MASK         = 0x7 << SX1280_STATUS_MODE_SHIFT,
    SX1280_STATUS_MODE_STDBY_RC     = 0x2 << SX1280_STATUS_MODE_SHIFT,
    SX1280_STATUS_MODE_STDBY_XOSC   = 0x3 << SX1280_STATUS_MODE_SHIFT,
    SX1280_STATUS_MODE_FS           = 0x4 << SX1280_STATUS_MODE_SHIFT,
    SX1280_STATUS_MODE_Rx           = 0x5 << SX1280_STATUS_MODE_SHIFT,
    SX1280_STATUS_MODE_Tx           = 0x6 << SX1280_STATUS_MODE_SHIFT,
};
```

Reply:
Should all those 1280's be 127x?

Reply:
Do we need to do that though? Adding another SPI transfer before we process the packet and doesn't add any benefit because in the ISR we're going to flip the mode anyway.

Reply:
I don't seem to be able to find that in the datasheet anywhere!

Reply:
> Do we need to do that though? Adding another SPI transfer before we process the packet adds a little latency and doesn't add any benefit because in the ISR we're going to flip the mode anyway.

Not if theres an extra SPI command!

Reply:
Sorry! Wrong radio... 🤦 Ignore.

Reply:
@CapnBry that status comes free, but for sx1280 :/